specific/cmo_hr/models/hr.py

```diff
@@ -0,0 +1,169 @@
+            'confirm': [('readonly', False)],
+        },
+    )
+    payment_by = fields.Selection(
+        [('cash', 'Cash'),
+         ('cashier_cheque', 'Cashier Cheque'),
+         ('bank_transfer', 'Bank Transfer'),
+         ('ac_payee', 'A/C Payee'),
+        ],
+        string='Payment By',
+    )
+    bank_transfer_ref = fields.Text(
+        string='Bank Transfer Ref.',
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    ac_payee_ref = fields.Text(
+        string='A/C Payee Ref.',
+        states={
+            'draft': [('readonly', False)],
+            'confirm': [('readonly', False)],
+        },
+    )
+    state = fields.Selection(
+        selection=[
+            ('draft', 'New'),
+            ('validate', 'Waiting Validate'),
+            ('cancelled', 'Refused'),
+            ('confirm', 'Waiting Approval'),
+            ('accepted', 'Approved'),
+            ('done', 'Waiting Payment'),
+            ('paid', 'Paid'),
+        ],
+    )
+    employee_id = fields.Many2one(
+        states={
+            'draft': [('readonly', True)],
+            'confirm': [('readonly', True)],
+        },
+    )
+    department_id = fields.Many2one(
+        states={
+            'draft': [('readonly', True)],
+            'confirm': [('readonly', True)],
+        },
+    )
+
+    @api.model
+    def create(self, vals):
+        ctx = self._context.copy()
+        current_date = datetime.date.today()
+        fiscalyear_id = self.env['account.fiscalyear'].find(dt=current_date)
+        ctx["fiscalyear_id"] = fiscalyear_id
+        if (not ctx.get('is_employee_advance', False)) and \
+                ctx.get('number', '/') == '/':
+            vals['number'] = self.env['ir.sequence']\
+                .with_context(ctx).get('cmo.expense')
+        elif ctx.get('is_employee_advance', False) and \
+                ctx.get('number', '/') == '/':
+            vals['number'] = self.env['ir.sequence']\
+                .with_context(ctx).get('cmo.advance')
+        res = super(HrExpenseExpense, self).create(vals)
+        return res
+
+    @api.multi
+    @api.onchange('payment_by')
+    def _onchange_payment_by(self):
+        self.ensure_one()
+        self.bank_transfer_ref = False
+        self.ac_payee_ref = False
+
+    @api.multi
+    def action_validate(self):
+        res = self.write({'state': 'validate'})
+        return res
+
+    @api.model
+    def _default_employee_request_id(self):
+        Employee = self.env['hr.employee']
+        employee_request_id = Employee.search([('user_id', '=', self._uid), ])
+        return employee_request_id and employee_request_id[0].id or False
+
+    @api.onchange('employee_id')
+    def _onchange_hr_department(self):
+        self.department_id = self.employee_id.department_id
+
+    @api.multi
+    @api.constrains('request_date', 'due_date')
+    def _check_due_date(self):
+        self.ensure_one()
```

Comment:
Do not use ensure_one(). Constrains should be multi.
Please use **for rec in self**.